kbase-extension/static/kbase/js/widgets/narrative_core/kbaseAppCard.js

```diff
@@ -0,0 +1,145 @@
+            var favorite = entry.app.favorite;
+            var app = entry.app.info;
+            
+            var shortName = entry.name ? entry.name : app.name;
+            var authors = entry.createdBy; 
+            var type = entry.version ? entry.version : ('v' + app.ver);
+
+            if(entry.createdBy === undefined){
+                if(app.authors.length >2){
+                    authors = app.authors.slice(0,2).join(', ') + '+ ' + (app.authors.length-2) + ' more';
+                }else{
+                    authors = app.authors.join(', ');
+                }
+            }
+            if (app.module_name && (entry.version === undefined)) {
+                type = '<a href="' + self.options.moduleLink + '/' + app.module_name + '" target="_blank">' +
+                    app.namespace + '</a> ' + type;
+            }
+
+            var $star = $('<i>').addClass('fa fa-star kbcb-star-default');
+            if (favorite) {
+                $star.addClass('fa fa-star kbcb-star-favorite').append('&nbsp;');
+            }
+
+            $star.click(function(e){
+                e.stopPropagation();
+                var params = {};
+                if (app.module_name) {
+                    params['module_name'] = app.module_name;
+                    params['id'] = app.id.split('/')[1];
+                } else {
+                    params['id'] = app.id;
+                }
+
+                if (favorite) {
+                    Promise.resolve(self.catalog.remove_favorite(params))
+                        .then(function () {
+                            $star.removeClass('kbcb-star-favorite');
+                            app.favorite = null; // important to set this if we don't refresh the panel
+                        })
+                        .catch(function (error) {
+                            console.error(error);
+                        });
+                } else {
+                    Promise.resolve(self.catalog.add_favorite(params))
+                        .then(function () {
+                            $star.addClass('kbcb-star-favorite');
+                            app.favorite = new Date().getTime(); // important to set this if we don't refresh the panel
+                        })
+                        .catch(function (error) {
+                            console.error(error);
+                        });
+                }            
+            })
+                .tooltip({
+                    title: 'Add or remove from your favorites',
+                    container: 'body',
+                    placement: 'bottom',
+                    delay: {
+                        show: Config.get('tooltip').showDelay,
+                        hide: Config.get('tooltip').hideDelay
+                    }
+                });
+
+            var $logo = $('<div>');
+            if (app.icon && app.icon.url) {
+                var url = self.options.methodStoreURL.slice(0, -3) + app.icon.url;
```

Comment:
You're already including Config here. You can get this url directly from:
`Config.url('narrative_method_store_image')` without having to slice it. It's set in `src/config.json`